lib/active_merchant/billing/affirm.rb

```diff
@@ -0,0 +1,49 @@
+require 'affirm'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class Affirm < Gateway
+      self.supported_countries = %w(US)
+      self.default_currency = 'USD'
+      self.money_format = :cents
+
+      # server can be :sandbox, or :production
+      def initialize(options = {})
+        requires!(options, :public_api_key, :private_api_key)
+        server = options[:test_mode] ? :sandbox : :production
+        ::Affirm.configure do |config|
+          config.public_api_key  = options[:public_api_key]
+          config.private_api_key = options[:private_api_key]
+          config.environment     = server
+        end
+        super
+      end
+
+      # purchase will be called when auto_capture is set to true
+      # it will do a authorize and a capture.
+      def purchase(money, affirm_source, options = {})
```

Comment:
If the gateway doesn't support purchase, don't implement it. Our code should call authorize when it wants to authorize, and capture when we want to capture. We dont want this to do both things, and ignore the response or errors from authorisation.